src/main/java/uk/gov/hmcts/cmc/claimstore/rules/ClaimantRepaymentPlanRule.java

```diff
@@ -0,0 +1,123 @@
+            case INSTALMENTS:
+                isPaymentPlanDateValidAgainstDefendantPlan(proposedPlan, getRepaymentPlanForResponse(response));
+                break;
+            case BY_SPECIFIED_DATE:
+                isPaymentPlanStartDateValidAgainstSetDate(proposedPlan);
+                break;
+            default:
+                throw new ClaimantInvalidRepaymentPlanException(
+                    INVALID_DEFENDANT_REPAYMENT_TYPE);
+        }
+    }
+
+
+    private void isPaymentPlanDateValidAgainstDefendantPlan(RepaymentPlan claimantPlan, RepaymentPlan defendantPlan) {
+        if (claimantPlan == null) {
+            throw new ClaimantInvalidRepaymentPlanException(EXPECTED_REPAYMENT_PLAN_CLAIMANT);
+        }
+
+        if (defendantPlan == null) {
+            throw new ClaimantInvalidRepaymentPlanException(EXPECTED_REPAYMENT_PLAN_DEFENDANT);
+        }
+
+        LocalDate defendantStartDate = defendantPlan.getFirstPaymentDate();
+        LocalDate monthFromNow = CalculateMonthIncrement.calculateMonthlyIncrement(LocalDate.now());
+        LocalDate thresholdDate = defendantStartDate.isBefore(monthFromNow) ? defendantStartDate : monthFromNow;
+
+        if (claimantPlan.getFirstPaymentDate().isBefore(thresholdDate)) {
+            throw new ClaimantInvalidRepaymentPlanException(
+                String.format(INSTALLMENT_DATE_MUST_BE_AFTER, thresholdDate));
+        }
+    }
+
+    private void isPaymentPlanStartDateValidAgainstSetDate(RepaymentPlan claimantPlan) {
+        if (claimantPlan == null) {
+            throw new ClaimantInvalidRepaymentPlanException(EXPECTED_REPAYMENT_PLAN_CLAIMANT);
+        }
+
+        LocalDate oneMonthFromNow = CalculateMonthIncrement.calculateMonthlyIncrement(LocalDate.now());
+        if (claimantPlan.getFirstPaymentDate().isBefore(oneMonthFromNow)) {
+            throw new ClaimantInvalidRepaymentPlanException(INSTALLMENT_DATE_MUST_BE_ONE_MONTH);
+        }
+    }
+
+    private PaymentOption getPaymentOptionForResponse(Response response) {
+        switch (response.getResponseType()) {
+            case PART_ADMISSION:
+                return ((PartAdmissionResponse) response).getPaymentIntention()
+                    .orElseThrow(() -> new ClaimantInvalidRepaymentPlanException(
+                        EXPECTED_PAYMENT_INTENTION))
+                    .getPaymentOption();
+            case FULL_ADMISSION:
+                return ((FullAdmissionResponse) response).getPaymentIntention().getPaymentOption();
+            default:
+                throw new ClaimantInvalidRepaymentPlanException(
+                    INVALID_RESPONSE_TYPE);
+        }
+    }
+
+    private RepaymentPlan getRepaymentPlanForResponse(Response response) {
+        switch (response.getResponseType()) {
+            case PART_ADMISSION:
+                return ((PartAdmissionResponse) response)
+                    .getPaymentIntention()
+                    .orElseThrow(() -> new ClaimantInvalidRepaymentPlanException(
```

Comment:
As payment plan here is for Defendant and name `ClaimantInvalidRepaymentPlanException` is misleading. Please rename the Exception class to `InvalidRepaymentPlanException` and then use here.

Reply:
The exception is only raised as a result of the start date of the claimant suggested repayment plan not meeting validation criteria, the defendant plan cannot trigger this exception.

Reply:
`((PartAdmissionResponse) response)` here from `Defendant` and not from Claimant.